planning/interpreters/filecoinv1.go

```diff
@@ -0,0 +1,68 @@
+package interpreters
+
+import (
+	"errors"
+
+	"github.com/filecoin-project/indexer-reference-provider/metadata"
+	v0 "github.com/filecoin-project/storetheindex/api/v0"
+	"github.com/ipfs-shipyard/w3rc/contentrouting"
+	"github.com/ipfs-shipyard/w3rc/planning"
+	"github.com/ipfs-shipyard/w3rc/planning/policies"
+)
+
+var _ planning.RoutingRecordInterpreter = (*FilecoinV1RecordInterpreter)(nil)
+
+type FilecoinV1RecordInterpreter struct {
+}
+
+func (fri FilecoinV1RecordInterpreter) Interpret(record contentrouting.RoutingRecord) (planning.PolicyScorer, error) {
+
+	// decode the record (or error) -- use metadata from filecoin
+	// check for free or paid policy
+	// return PolicyResults that when given "prefer_free" returns 1 if retrieval is free or zero if its paid
+
+	if record.Protocol() == contentrouting.RoutingErrorProtocol {
+		err, ok := record.Payload().(error)
+		if !ok {
+			return nil, errors.New("routing record payload not match expected type: error")
```

Comment:
```suggestion
			return nil, errors.New("routing record payload does not match expected type: error")
```

Spelling mistake from my previous PR; apologies